src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -111,24 +113,28 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig {
     logger.info("Processing Tezos Blocks..")
 
     val blocksToSynchronize = tezosConf.depth match {
-      case Everything => tezosNodeOperator.getAllBlocks(tezosConf.network)
       case Newest => tezosNodeOperator.getBlocksNotInDatabase(tezosConf.network)
-      case Custom(n) => tezosNodeOperator.getLatestBlocks(tezosConf.network, n)
+      case Everything => tezosNodeOperator.getLatestBlocks(tezosConf.network)
+      case Custom(n) => tezosNodeOperator.getLatestBlocks(tezosConf.network, Some(n))
     }
 
     blocksToSynchronize.flatMap {
       blocksWithAccounts =>
-        db.run(TezosDb.writeBlocksAndCheckpointAccounts(blocksWithAccounts.toMap))
-          .andThen {
-            case Success(accountsCount) =>
-              logger.info("Wrote {} blocks to the database, checkpoint stored for{} account updates", blocksWithAccounts.size, accountsCount.fold("")(" " + _))
-            case Failure(e) =>
-              logger.error(s"Could not write blocks or accounts checkpoints to the database because $e")
-          }
-          .map(_ => Done)
-    }.andThen {
+
+        def logOutcome[A](outcome: Future[Option[A]]): Future[Option[A]] = outcome.andThen {
```

Comment:
Having the `andThen` to log the outcome of the operation was cluttering the code flow.
I extracted the logging effect into this wrapping function, so that the logic was separated from the main operational flow